### PR TITLE
gh-116782: Mention `__type_params__` in `inspect.getmembers` docs

### PR DESCRIPTION
**Pull request by sobolevn**
* Issue: gh-116782

----
📚 Documentation preview 📚: https://cpython-previews--116783.org.readthedocs.build/

### CODE REVIEW

File: Doc/library/inspect.rst [hunk 1]
```diff
@@ -55,6 +55,10 @@ attributes (see :ref:`import-mod-attrs` for module attributes):
 |           | __module__        | name of module in which   |
 |           |                   | this class was defined    |
 +-----------+-------------------+---------------------------+
+|           | __type_params__   | :pep:`695` type           |
```
Comment (JelleZijlstra) [inline Doc/library/inspect.rst]: We shouldn't link to PEPs as long-term documentation. (Concrete reason: PEP 696 will soon change these again, and PEP 695 won't mention it.) Maybe link to https://docs.python.org/3.12/reference/compound_stmts.html#type-params ?

File: Doc/library/inspect.rst [hunk 2]
```diff
@@ -55,9 +55,10 @@ attributes (see :ref:`import-mod-attrs` for module attributes):
 |           | __module__        | name of module in which   |
 |           |                   | this class was defined    |
 +-----------+-------------------+---------------------------+
-|           | __type_params__   | :pep:`695` type           |
-|           |                   | parameters of generic     |
-|           |                   | classes                   |
+|           | __type_params__   | A tuple containing the    |
+|           |                   | :ref:`type <type-params>` |
+|           |                   | parameters of             |
```
Comment (JelleZijlstra) [inline Doc/library/inspect.rst]:
```suggestion
|           |                   | :ref:`type parameters     |
|           |                   | <type-params>` of         |
```
Better to link the whole phrase

File: Doc/library/inspect.rst [hunk 3]
```diff
@@ -107,9 +108,10 @@ attributes (see :ref:`import-mod-attrs` for module attributes):
 |           |                   | reserved for return       |
 |           |                   | annotations.              |
 +-----------+-------------------+---------------------------+
-|           | __type_params__   | :pep:`695` type           |
-|           |                   | parameters of generic     |
-|           |                   | functions                 |
+|           | __type_params__   | A tuple containing the    |
+|           |                   | :ref:`type <type-params>` |
+|           |                   | parameters of             |
```
Comment (JelleZijlstra) [inline Doc/library/inspect.rst]:
```suggestion
|           |                   | :ref:`type parameters     |
|           |                   | <type-params>` of         |
```

**Review by JelleZijlstra — APPROVED** [review]

**Comment by sobolevn** [comment]
Thanks!

**Pull request merged**